forge-game/src/main/java/forge/game/combat/CombatUtil.java

```diff
@@ -288,8 +290,14 @@ public static boolean checkPropagandaEffects(final Game game, final Card attacke
         // If there's a better way of handling this somewhere deeper in the code, feel free to remove
         final SpellAbility fakeSA = new SpellAbility.EmptySa(attacker, attacker.getController());
         fakeSA.setCardState(attacker.getCurrentState());
-        return attacker.getController().getController().payManaOptional(attacker, attackCost, fakeSA,
-                "Pay additional cost to declare " + attacker + " an attacker", ManaPaymentPurpose.DeclareAttacker);
+
+        attacker.getController().getManaPool().restoreColorReplacements();
```

Comment:
the problem is it might be needed for human
but for AI both `payManaOptional` and `payCostToPreventEffect` will lead to `ComputerUtilMana.payManaCost` anyway

so we either need to pass the matrix a lot deeper so it knows it can avoid duplicate work or move this somehow?

Reply:
i'm trying to check where we would need to apply them deeper for AI or not

or if we should change that for Human

@tehdiplomat your opinions?

Reply:
Well I think I cleaned it up, also less code needed this way